src/EnergyPlus/OutputProcessor.cc

```diff
@@ -6823,6 +6819,29 @@ void UpdateMeterReporting(EnergyPlusData &state)
         ErrorsLogged = true;
     }
 
+    // Helper lambda to locate a meter index from its name. Returns a negative value if not found
+    auto findMeterIndexFromMeterName = [](std::string const &name) -> int {
+        // Return a value <= 0 if not found
+        int meterIndex = -99;
+
+        std::string::size_type wildCardPosition = index(name, '*');
+
+        if (wildCardPosition == std::string::npos) {
+            meterIndex = UtilityRoutines::FindItem(name, OutputProcessor::EnergyMeters);
+        } else { // Wildcard input
+            for (int Meter = 1; Meter <= OutputProcessor::NumEnergyMeters; ++Meter) {
+                if (UtilityRoutines::SameString(OutputProcessor::EnergyMeters(Meter).Name.substr(0, wildCardPosition),
+                                                name.substr(0, wildCardPosition)))
+                {
+                    meterIndex = Meter;
+                    break;
+                }
+            }
+        }
+
+        return meterIndex;
+    };
+
```

Comment:
I initially went for a solution were I was reusing this routine (among with checking if it was an output:variable or a Schedule) so I made it its own function and dried up the code below as well. I'm not using it, and it's not going to be used anywhere else, so I made it a lambda.